### PR TITLE
Implement offsetToZero() function

### PR DESCRIPTION
**Pull request by colega**
Implementing offsetToZero() function, which is very similar to offset(),
but we just need to find the minimum value first. NaN and infinities are
skipped, as they'll just keep being infinities and NaNs after
offsetting.

Signed-off-by: Oleg Zaytsev <mail@olegzaytsev.com>

### CODE REVIEW

**Review by ywwg — DISMISSED** [review]

File: expr/func_offsettozero_test.go [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+		[]models.Series{
+			getSeriesNamed("offsetToZero(a)", out),
+		},
+		t,
+	)
+}
+
+func TestOffsetToZeroMultiple(t *testing.T) {
+	const offset = 10
+	in := []schema.Point{
```
[the remark this answers is not in the document]
Comment (Dieterbe) [reply expr/func_offsettozero_test.go]: much of the testing code stems from years ago, we can probably improve the style. But that's a separate conversation, and may not be the best use of our time.
The change to table driven tests is fine, and I think we already do that in a bunch of places anyway.

File: expr/func_offsettozero_test.go [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+			input: []models.Series{
+				getSeriesNamed("a", []schema.Point{
+					{Val: 0 + positiveOffset, Ts: 10},
+					{Val: 0 + positiveOffset, Ts: 20},
+					{Val: 5.5 + positiveOffset, Ts: 30},
+					{Val: math.NaN(), Ts: 40},
+					{Val: math.NaN(), Ts: 50},
+					{Val: 1234567890 + positiveOffset, Ts: 60},
+				}),
+				getSeriesNamed("b", []schema.Point{
```
Comment (Dieterbe) [inline expr/func_offsettozero_test.go:108]: maybe use different numbers for `b` to make the test more specific and robust? (e.g. to check that a and b don't get mixed up)

**Review by Dieterbe — APPROVED** [review]
> Implementing offsetToZero() function, which is very similar to offset(),
but we just need to find the minimum value first. NaN and infinities are
skipped, as they'll just keep being infinities and NaNs after
offsetting.

awesome. thanks Oleg!
This PR looks good to me. just left you a comment about the tests

File: expr/func_offsettozero_test.go (hunk 2 above)
Comment (colega) [reply expr/func_offsettozero_test.go:108]: Makes sense, done in https://github.com/grafana/metrictank/pull/1995/commits/c3cce88d1c25917635e84a9e8345366bd88f9e1e

**Pull request merged**